sheetstorm/management/commands/generate_test_data.py

```diff
@@ -61,6 +75,42 @@ def _set_user_factory_parameters(user_type):
             "is_superuser": user_type == superuser_user_type,
         }
 
+    @staticmethod
+    def _get_project_options(options):
+        return {
+            suspended_project: options[suspended_project] - Project.objects.filter_suspended().count()
+            if options[suspended_project] is not None
+            else 0,
+            active_project: options[active_project] - Project.objects.filter_active().count()
+            if options[active_project] is not None
+            else 0,
+            completed_project: options[completed_project] - Project.objects.filter_completed().count()
+            if options[completed_project] is not None
+            else 0,
+        }
+
+    def create_project(self, project_type, number_of_projects_to_create):
+        factory_parameters = self._set_project_factory_parameters(project_type)
+
+        for project_number in range(number_of_projects_to_create):
+            logging.info(f"{number_of_projects_to_create - project_number} {project_type} project(s) left to create")
+            ProjectFactory(**factory_parameters)
+
+    def _set_project_factory_parameters(self, project_type):
+        return {
+            "start_date": self._set_project_start_date(timezone.now()),
+            "suspended": project_type == suspended_project,
+            "stop_date": self._set_project_stop_date(timezone.now()) if project_type == completed_project else None,
+        }
+
+    @staticmethod
+    def _set_project_start_date(date):
+        return date + relativedelta(months=-1, day=1)
```

Comment:
It will be problematic that you're giving here date and from this date you're subtract period of time. It should be done differently - from current date you should subtract period of time which should be has default value but also should be modifiable.

Reply:
Also if you want to subtract date don't add negative date. Just subtract positive date and it will also work

Reply:
Done